src/main/java/com/google/maps/DirectionsApiRequestBase.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2015 Google Inc. All rights reserved.
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.google.maps;
+
+import com.google.maps.internal.ApiConfig;
+import com.google.maps.internal.ApiResponse;
+import com.google.maps.model.LatLng;
+import com.google.maps.model.TrafficModel;
+import com.google.maps.model.TransitMode;
+import com.google.maps.model.TransitRoutingPreference;
+import com.google.maps.model.TravelMode;
+import com.google.maps.model.Unit;
+import org.joda.time.ReadableInstant;
+
+import static com.google.maps.internal.StringJoin.join;
+
+/**
+ * Common base class for DirectionsApiRequest and FullDirectionsApiRequest.
```

Comment:
It's not that the requests are different, it's that the returned data is marshalled differently. In one case we return just the route, while in the other we return route an geocoded waypoints.